UnitsNet/BaseDimensions.cs

```diff
@@ -31,7 +31,22 @@ public BaseDimensions(int length, int mass, int time, int current, int temperatu
         public bool IsBaseQuantity()
         {
             var dimensionsArray = new[] { Length, Mass, Time, Current, Temperature, Amount, LuminousIntensity };
-            bool onlyOneEqualsOne = dimensionsArray.Where(dimension => dimension == 1).Take(2).Count() == 1;
+            bool onlyOneEqualsOne = false;
+            foreach (var dimension in dimensionsArray)
+            {
+                if (1 == dimension)
+                {
+                    if (onlyOneEqualsOne)
+                    {
+                        return false;
+                    }
+                    onlyOneEqualsOne = true;
+                }
+                else if (0 != dimension)
+                {
+                    return false;
+                }
+            }
```

Comment:
Here is a linq version

```cs
        public bool IsBaseQuantity()
        {
            var dimensions = new[] { Length, Mass, Time, Current, Temperature, Amount, LuminousIntensity };
            return dimensions.Take(2).Count(x => x == 1) == 1 &&
                   dimensions.All(x => x is 0 or 1);
        }

```

Reply:
Thanks for the suggestion!
I found single pass `Select` + `Sum` approach